plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbClient.java

```diff
@@ -617,4 +652,236 @@ private static Optional<ColumnMapping> getUnsignedMapping(JdbcTypeHandle typeHan
+                    }
+                    else {
+                        // distinct values count has value only when histogram_type=JSON_HB,
+                        // but having distinct values count will make PushedDown useless,so set row count
+                        columnStatisticsBuilder.setDistinctValuesCount(Estimate.of(rowCount));
+                    }
+                }
+
+                Type type = column.getColumnType();
+                // MariaDB Date, Time and Timestamp also have the min and max value
+                // But currently, we only consider the numbers
+                if (type == TINYINT ||
+                        type == SMALLINT ||
+                        type == INTEGER ||
+                        type == BIGINT ||
+                        type == REAL ||
+                        type == DOUBLE ||
+                        type instanceof DecimalType) {
+                    // low_value and high_value
+                    if (result.getMinValue() != null && result.getMaxValue() != null) {
+                        DoubleRange range = new DoubleRange(Double.parseDouble(result.getMinValue()), Double.parseDouble(result.getMaxValue()));
+                        columnStatisticsBuilder.setRange(range);
+                    }
+                }
+
+                tableStatistics.setColumnStatistics(column, columnStatisticsBuilder.build());
+            }
+            return tableStatistics.build();
+        }
+    }
+
+    private static class StatisticsDao
+    {
+        private final Handle handle;
+
+        public StatisticsDao(Handle handle)
+        {
+            this.handle = requireNonNull(handle, "handle is null");
+        }
+
+        Long getRowCount(JdbcTableHandle table)
+        {
+            return handle.createQuery("" +
+                            "SELECT cardinality FROM mysql.table_stats " +
+                            "WHERE db_name = :schema AND table_name = :table_name ")
+                    .bind("schema", table.getCatalogName())
+                    .bind("table_name", table.getTableName())
+                    .mapTo(Long.class)
+                    .findOne()
+                    .orElse(null);
+        }
+
+        List<ColumnStatisticsResult> getColumnStatistics(JdbcTableHandle table)
+        {
+            try {
+                handle.execute("SELECT 1 FROM mysql.column_stats WHERE 0=1");
+            }
+            catch (UnableToExecuteStatementException e) {
+                if (nullToEmpty(e.getMessage()).contains("Unknown table 'mysql.column_stats'")) {
+                    return new ArrayList<>();
+                }
+            }
+
+            return handle.createQuery("" +
+                            "SELECT column_name, avg_length, min_value, max_value, nulls_ratio, hist_type, " +
+                            "decode_histogram(hist_type, histogram) as histogram FROM mysql.column_stats " +
```

Comment:
nit: Move to `decode_histogram(hist_type, histogram) AS histogram` to above line and uppercase `AS`.

Reply:
Done